File: runtime/cpp/titan-ast-runtime-lib/SyntaxDfaState.cpp
//
// Created by tian wei jun on 2022/12/1 0001.
//

#include "SyntaxDfaState.h"

SyntaxDfaState::SyntaxDfaState(int index)
    : index(index), type(0),
      edges(std::unordered_map<const Grammar *, SyntaxDfaState *,
                               PtrGrammarContentHash, PtrGrammarContentEq>()),
      closingProductionRules(std::vector<ProductionRule *>()) {}

// edges grammar delete by PersistentData.gammars
// edges SyntaxDfaState delete by dfa
// closingProductionRules delete by  PersistentData.productionRules
SyntaxDfaState::~SyntaxDfaState() = default;

File: runtime/java/titan-ast-runtime/src/titan/ast/runtime/SyntaxDfaState.java
package titan.ast.runtime;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

/**
 * 语法对应的 确定有限状态自动机 的状态.
 *
 * @author tian wei jun
 */
class SyntaxDfaState {
  int index = 0;
  int type = FaStateType.NONE.getValue();
  // 转移
  Map<Grammar, SyntaxDfaState> edges = new HashMap<>();
  ArrayList<ProductionRule> closingProductionRules = new ArrayList<>();

  SyntaxDfaState(int index) {
    this.index = index;
  }
}
